extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -153,63 +163,103 @@ export class SubscriptionService {
   }
 
   /**
-   * @param {!Entitlements} entitlements
+   * @param {!Entitlement} entitlement
    * @private
    */
-  resolveEntitlementsToStore_(entitlements) {
-    this.entitlementStore_.resolveEntitlement(entitlements.service,
-        entitlements);
+  resolveEntitlementsToStore_(entitlement) {
+    this.entitlementStore_.resolveEntitlement(entitlement.service, entitlement);
   }
 
   /**
    *
    * @param {!SubscriptionPlatform} subscriptionPlatform
    */
   fetchEntitlements_(subscriptionPlatform) {
-    subscriptionPlatform.getEntitlements().then(entitlements =>
-      this.resolveEntitlementsToStore_(entitlements));
+    subscriptionPlatform.getEntitlements().then(entitlement =>
+      this.resolveEntitlementsToStore_(entitlement));
   }
 
   /**
-   * @return {!SubscriptionService}
-   * @private
+   * @return {!Promise}
    */
-  start_() {
-    this.initialize_().then(() => {
+  start() {
+    return this.initialize_().then(() => {
+
       this.renderer_.toggleLoading(true);
 
       user().assert(this.pageConfig_, 'Page config is null');
 
-      user().assert(this.serviceConfig_['services'],
+      user().assert(this.platformConfig_['services'],
           'Services not configured in service config');
 
-      this.serviceConfig_['services'].forEach(service => {
-        this.initializeLocalPlatforms_(service,
-            /** @type {!PageConfig} */(this.pageConfig_));
-      });
-
-      const serviceIds = this.serviceConfig_['services'].map(service =>
+      const serviceIds = this.platformConfig_['services'].map(service =>
         service['serviceId'] || 'local');
 
       this.entitlementStore_ = new EntitlementStore(serviceIds);
 
+      this.platformConfig_['services'].forEach(service => {
+        this.initializeLocalPlatforms_(service);
+      });
+
       this.subscriptionPlatforms_.forEach(subscriptionPlatform => {
         this.fetchEntitlements_(subscriptionPlatform);
       });
 
       this.entitlementStore_.getGrantStatus()
           .then(grantState => {this.processGrantState_(grantState);});
 
-      this.entitlementStore_.selectPlatform()
-          .then(entitlements => {
-            this.subscriptionPlatforms_.forEach(platform => {
-              if (platform.getServiceId() == entitlements.service) {
-                platform.activate();
-              }
-            });
-          });
+      this.selectAndActivatePlatform_();
+    });
+  }
+
+  selectAndActivatePlatform_() {
+    this.entitlementStore_.selectPlatform().then(entitlement => {
+      /** @type {!RenderState} */
+      const renderState = {
+        entitlement,
+        loggedIn: false, // TODO (@prateekbh): ask how to derive this?
+        subscribed: false, // TODO (@prateekbh): ask how to derive this?
```

Comment:
This should be `subscribed = !!entitlement.subsciptionToken`. I.e. presense of subscriptionToken indicates subscriber.